Add Vendor as Key Field for CommonServerPython's Endpoint

MichaelYochpaz opened the pull request:
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Description
This PR adds the `Vendor` field as a key when comparing Endpoint objects if it exists.

yaakovi reviewed (commented):
* Why did you modify: `Packs/ApiModules/Scripts/CoreIRApiModule/CoreIRApiModule.py`? revert if not needed.
* add release notes to the `Base` pack

MichaelYochpaz commented:
@yaakovi I had flake8 linting errors for the `CoreIRApiModule` pack, but it seems that it was only the test tile that required a fix to pass the CI linting step.
I've removed the changes and added release notes.
